### PR TITLE
fix: check cloud init status

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

hotfix: check cloud-init status to see if job is already done before health check.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/openstack_cloud/openstack_manager.py [hunk 1]
```diff
@@ -532,20 +532,33 @@ def _ssh_health_check(conn: OpenstackConnection, server_name: str, startup: bool
...
-        logger.debug("Output of `ps aux` on %s stderr: %s", server_name, result.stderr)
-        if not result.ok:
+        cloud_init_result: invoke.runners.Result = ssh_conn.run("cloud-init status", warn=True)
+        logger.debug(
+            "Output of `cloud-init status` on %s stderr: %s", server_name, cloud_init_result.stderr
+        )
+        if not cloud_init_result.ok:
+            logger.warning("cloud-init status command failed on %s.", server_name)
+            raise _SSHError(f"cloud-init status command failed on {server_name}.")
+        # cloud-init status should be "running"
```
Comment (mthaddon) [inline src/openstack_cloud/openstack_manager.py:542]: Based on this comment I'd expect the line below to be `if "running" not in cloud_init_result.stdout`. What do you think?

**Comment by jdkandersson** [comment]
Do you need to fix the target branch?

**Comment by yanksyoon** [comment]
> Do you need to fix the target branch?
> 
> 

What do you mean by fixing the target branch please?

**Comment by jdkandersson** [comment]
I'm guessing this isn't needed anymore?

**Comment by yanksyoon** [comment]
@jdkandersson Yup, thanks for the comment, closing!

**Pull request closed without merging**